src/opendr/perception/speech_transcription/vosk/vosk_learner.py

```diff
@@ -0,0 +1,382 @@
+        accept_waveform = self.rec.AcceptWaveform(byte_data)
+        if accept_waveform:
+            output = self.rec.Result()
+            text = json.loads(output)["text"]
+        else:
+            output = self.rec.PartialResult()
+            text = json.loads(output)["partial"]
+
+        return VoskTranscription(text=text, accept_waveform=accept_waveform)
+
+    def _preprocess(self, data: Union[np.ndarray, bytes]) -> bytes:
+        """
+        Convert audio data to bytes.
+
+        Args:
+            data (Union[np.ndarray, bytes]): Audio data
+
+        Returns:
+            bytes: Audio data in bytes.
+        """
+        if isinstance(data, bytes):
+            return data
+
+        # Convert the array to int16, as vosk expects 16-bit integer data.
+        data = (data * np.iinfo(np.int16).max).astype(np.int16).tobytes()
+
+        return data
+
+    def save(self):
+        return
+
+    def reset(self):
+        """
+        Set Vosk model, model name, language, and KalidRecognizer to None. Use before loading a new model.
+        """
+        self.model_name = None
+        self.language = None
+        self.model = None
+        self.rec = None
+
+    def reset_rec(self):
+        """
+        Reset the KalidRecognizer.
+        """
+        assert self.rec is not None, "KalidRecognizer is not loaded. Please the load() method before resetting."
+        self.rec.Reset()
+
+    def fit(self):
+        return
+
+    def eval(self, dataset: DatasetIterator, save_path_csv: Optional[str] = None) -> Dict:
+        """
+        Evaluate Vosk model on the given dataset.
+
+        Args:
+            dataset (DatasetIterator): A speech dataset.
+            save_path_csv (str, optional): The path to save the evaluation results.
+
+        Returns:
+            Dict: A dictionary containing the word error rate (WER).
+
+        Raises:
+            AssertionError: If the model is not loaded.
+        """
+
+        assert self.model is not None and self.rec is not None, "Model and KaldiRecognizer is not loaded. Please load a model before evaluating."
+
+        loader = DataLoader(
+            dataset,
+            batch_size=1,
+            shuffle=False,
+            drop_last=False,
+        )
+
+        hypotheses = []
+        references = []
+
+        for audio, text in tqdm(loader):
+            # Remove batch dimension
+            audio = audio[0]
+            text = text[0]
+
+            result = self.infer(audio)
+            self.reset_rec()
+
+            hypotheses.append(result.text.lower())
+            references.append(text.lower())
+
+        transcriptions = pd.DataFrame(dict(hypothesis=hypotheses, reference=references))
+        if save_path_csv is not None:
+            transcriptions.to_csv(save_path_csv, index=False)
+
+        wer = jiwer.wer(list(transcriptions["reference"]), list(transcriptions["hypothesis"]))
+
+        return {"wer": wer}
+
+    def optimize(self):
+        return
```

Comment:
For methods that are not implement usually this is used (see for example the efficient_ps_learner.py).
Could you please update all the occurrences in your new learners?
```suggestion
        """This method is not used in this implementation."""
        raise NotImplementedError
```